extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -285,6 +303,49 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
         this.win, DOUBLECLICK_EXPERIMENT_FEATURE.DELAYED_REQUEST);
   }
 
+  /** @override */
+  buildCallback() {
+    super.buildCallback();
+    if (this.win['dbclk_a4a_viz_change']) {
```

Comment:
Overall, I'd prefer an ampdoc service b/c (a) it already supports one-per-doc logic,  (b) it extends well to PWA/shadow docs, and (c) could be a good placeholder for other state instead of adding more stuff to global window (e.g. `ampCorrelator`).